common/src/types/token.rs

```diff
@@ -668,6 +697,11 @@ impl Token {
         Self::from_starknet_goerli_addr(&format!("0x{}", addr.to_str_radix(16 /* radix */)))
     }
 
+    /// Given an ERC-20 contract address on Katana (represented as a `BigUint`) returns a Token
+    pub fn from_katana_addr_biguint(addr: &BigUint) -> Self {
+        Self::from_katana_addr(&format!("0x{}", addr.to_str_radix(16 /* radix */)))
```

Comment:
can we break the bigint -> string out into a private helper now that a few methods use it, specifically:
```rust
fn biguint_to_str_addr(x: &BigUint) -> String {
    let addr = x.to_str_radix(16 /* radix */);
    format!("0x{addr}")
}
```